# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../tesseract/gcc41.patch
# Copyright (C) 2007 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

--- ./cutil/tordvars.h~ 2007-02-04 16:18:12.000000000 +0000
+++ ./cutil/tordvars.h  2007-02-04 16:19:07.000000000 +0000
@@ -39,6 +39,8 @@
 extern FILE *correct_fp;         //correct text
 extern FILE *matcher_fp;
 
+extern "C"
+{
 extern int blob_skip;            /* Skip to next selection */
 extern int num_word_choices;     /* How many words to keep */
 extern int similarity_enable;    /* Switch for Similarity */
@@ -49,6 +51,7 @@
 extern int show_bold;            /* Use bold text */
 extern int display_text;         /* Show word text */
 extern int display_blocks;       /* Show word as boxes */
+}
 
 extern float overlap_threshold;  /* Overlap Threshold */
 extern float certainty_threshold;/* When to quit looking */
--- ./cutil/globals.h~  2007-02-04 16:17:07.000000000 +0000
+++ ./cutil/globals.h   2007-02-04 16:17:54.000000000 +0000
@@ -43,9 +43,15 @@
 extern int debugs[MAXPROC];      /*debug flags */
 extern int plots[MAXPROC];       /*plot flags */
 extern int corners[4];           /*corners of scan window */
+#ifdef __cplusplus
+extern "C" {
+#endif
 extern int optind;               /*option index */
 extern char *optarg;             /*option argument */
                                  /*image file name */
+#ifdef __cplusplus
+}
+#endif
 extern char imagefile[FILENAMESIZE];
                                  /* main directory */
 extern char directory[FILENAMESIZE];
